testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.testutils;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.hub.HubChannel;
+import com.amplifyframework.hub.HubEvent;
+import com.amplifyframework.hub.HubEventFilter;
+import com.amplifyframework.hub.HubEventFilters;
+import com.amplifyframework.hub.SubscriptionToken;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Accumulates {@link HubEvent}s received on a {@link HubChannel}, into an in-memory
+ * buffer. These may be queried by tests in a synchronous way to check if they have arrived.
+ */
+@SuppressWarnings({"unused", "UnusedReturnValue"})
+public final class HubAccumulator {
+    private final HubChannel channel;
+    private final HubEventFilter filter;
```

Comment:
I'm not sure if we ever got the filter implemented fully in Hub on Android, just an FYI

Reply:
For Hub, we did, it's [more or less just a single `if (...)` check](https://github.com/aws-amplify/amplify-android/blame/e9536f2df8f7f7803faefa4cf5926103b351dab0/core/src/main/java/com/amplifyframework/hub/AWSHubPlugin.java#L66-L73), in the `publish(...)` method.